src/Microsoft.Azure.SignalR.Common/Interfaces/IServiceConnection.cs

```diff
@@ -15,6 +15,8 @@ internal interface IServiceConnection
 
         Task StopAsync();
 
+        Task CloseAsync(TimeSpan timeout);
```

Comment:
Is it Container level decision to `Shutdown`? and for the inner connection, it can just provide a method to `wait for all complete`. Looks like it is not neccessory for the inner connection to have a `Close` logic

Reply:
The problem is we hold `IServiceConnection` in our `ServiceConnectionContainer` so I have no other choice except for adding a method to the interface. Reflection/Conversion could also achieve this but it's not a good choice I guess.